edx_proctoring/views.py

```diff
@@ -478,6 +479,86 @@ def get(self, request, exam_id=None, course_id=None, content_id=None):  # pylint
         return Response(data)
 
 
+class RegisterProctoredExamsView(ProctoredAPIView):
+    """
+    Endpoint for registering all proctored exams in a course
+    """
+    @method_decorator(require_staff)
+    def patch(self, request, course_id):
+        """
+        Create or update a list of all active exams.
+        Exams not included in the registration payload will be deactivated
+        """
+        request_exams = request.data
+        write_result = []
+
+        serializer = ProctoredExamRegistrationSerializer(data=request_exams, many=True)
+
+        if serializer.is_valid():
+            course_exams = {exam.content_id: exam for exam in ProctoredExam.get_all_exams_for_course(str(course_id))}
+            for request_exam in request_exams:
+                content_id = request_exam.get('content_id')
+
+                if content_id in course_exams:
+                    exam = course_exams.pop(content_id)
+                    exam_id = update_exam(
+                        exam.id,
+                        exam_name=request_exam.get('exam_name', None),
+                        time_limit_mins=request_exam.get('time_limit_mins', None),
+                        due_date=request_exam.get('due_date', None),
+                        is_proctored=request_exam.get('is_proctored', None),
+                        is_practice_exam=request_exam.get('is_practice_exam', None),
+                        external_id=request_exam.get('external_id', None),
+                        is_active=request_exam.get('is_active', None),
+                        hide_after_due=request_exam.get('hide_after_due', None),
+                        backend=request_exam.get('backend', None),
+                    )
+
+                    write_result.append({'exam_id': exam.id})
+                    msg = f'Updated exam {exam.id}'
```

Comment:
do we want exam ID or the content_id which is the lookup key? Or both?

Reply:
personally, if I'm debugging anything I'm just going to look this up by the database id which is a whole lot cleaner than messing with a big string (content_id)